File: CodeCheck/Published Standards/SEI Cert Standards/Java/MET11-J/Noncompliant/Employee.java
// Mutable class Employee
class Employee_2 {
    private String name;
    private double salary;

    Employee_2(String empName, double empSalary) {
        this.name = empName;
        this.salary = empSalary;
    }

    public void setEmployeeName(String empName) {
        this.name = empName;
    }

    public void setSalary(double empSalary) {
        this.salary = empSalary;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Employee_2)) {
            return false;
        }

        Employee_2 emp = (Employee_2) o;
        return emp.name.equals(name);   //UndCC_Violation
    }

}

File: CodeCheck/Published Standards/SEI Cert Standards/C++/OOP51-CPP/Employee.cpp
#include "Employee.h"

Employee::Employee(const std::string &name) : name(name) {}
const std::string &Employee::get_name() const { return name; }

std::ostream &operator<<(std::ostream &os, const Employee &e)
{
    e.print(os);
    return os;
}

void Employee::print(std::ostream &os) const
{
    os << "Employee: " << get_name() << std::endl;
}
